### ISSUE TITLE
Gather should collect broker resources when broker kubeconfig is passed

### ISSUE THREAD

**Issue by Jaanki** [submariner-io/submariner-operator#1276]
**What happened**:
`subctl gather` output shows broker resources not found when borker kubeconfig is passed
```
Gathering information from cluster "cluster1"
 ✓ Gathering operator logs
 ✓ Found 1 pods matching label selector "name=submariner-operator"
 ✓ Gathering operator resources
 ✓ Found 0 submariners in namespace "submariner-operator"
 ✓ Found 0 servicediscoveries in namespace "submariner-operator"
 ✓ Found 1 deployments by field selector "metadata.name=submariner-operator" in namespace "submariner-operator"
 ✓ Found 0 daemonsets by label selector "app=submariner-gateway" in namespace "submariner-operator"
 ✓ Found 0 daemonsets by label selector "app=submariner-routeagent" in namespace "submariner-operator"
 ✓ Found 0 daemonsets by label selector "app=submariner-globalnet" in namespace "submariner-operator"
 ✓ Found 0 deployments by label selector "app=submariner-networkplugin-syncer" in namespace "submariner-operator"
 ✓ Found 0 deployments by label selector "app=submariner-lighthouse-agent" in namespace "submariner-operator"
 ✓ Found 0 deployments by label selector "app=submariner-lighthouse-coredns" in namespace "submariner-operator"
 ✓ Gathering connectivity logs
 ✓ Found 0 pods matching label selector "app=submariner-gateway"
 ✓ Found 0 pods matching label selector "app=submariner-routeagent"
 ✓ Found 0 pods matching label selector "app=submariner-globalnet"
 ✓ Found 0 pods matching label selector "app=submariner-networkplugin-syncer"
 ✓ Gathering connectivity resources 
 ✓ athering CNI data from 0 pods matching label selector "app=submariner-routeagent"
 ✓ Gathering CNI data from 0 pods matching label selector "app=submariner-gateway"
 ✓ Gathering cable driver data from 0 pods matching label selector "app=submariner-gateway"
 ✓ Found 0 endpoints in namespace "submariner-operator"
 ✓ Found 0 clusters in namespace "submariner-operator"
 ✓ Found 0 gateways in namespace "submariner-operator"
 ✓ Gathering service-discovery resources 
 ✓ Found 0 serviceexports in namespace ""
 ✓ Found 0 serviceimports in namespace ""
 ✓ Found 0 endpointslices by label selector "endpointslice.kubernetes.io/managed-by=lighthouse-agent.submariner.io" in namespace ""
 ✓ Found 0 configmaps by label selector "component=submariner-lighthouse" in namespace "submariner-operator"
 ✓ Found 1 configmaps by field selector "metadata.name=coredns" in namespace "kube-system"
 ✓ Gathering service-discovery logs
 ✓ Found 0 pods matching label selector "component=submariner-lighthouse"
 ✓ Found 2 pods matching label selector "k8s-app=kube-dns"
 ✗ Gathering broker resources
 ✗ Error getting the broker's rest config: error obtaining the Submariner resource: submariners.submariner.io "submariner" not found
```
cluster1 is broker cluster. All other module's information is successfully collected.

SImilarly for `$ subctl gather --module=broker
Gathering information from cluster "cluster1"
 ✗ Gathering broker resources
 ✗ Error getting the broker's rest config: error obtaining the Submariner resource: submariners.submariner.io "submariner" not found
`

**What you expected to happen**:
Broker resources to be collected

**How to reproduce it (as minimally and precisely as possible)**:
Run `subctl gather`

**Environment**:
- Submariner version (use `subctl version`): 0.9.0

**Comment by tpantelis** [comment]
So `cluster1` was only the broker, ie no connectivity or service discovery deployed. The error is fixed by #1283. I guess if we can't obtain the broker's rest config that way we could assume it is the broker cluster and use `info.RestConfig`. However we still wouldn't have the broker's namespace - we'd have to assume it's `submariner-k8s-broker.` The other option is to skip "_Gathering broker resources_" in this case (as will happen with #1283) so we don't make assumptions.  The broker info would be gathered by another data cluster anyway. If only the broker cluster was specified then it really wouldn't be useful to just gather the broker resources.

**Comment by nyechiel** [comment]
As per discussion, the Broker cluster could be detected by looking at the `broker` CR.
